### PR TITLE
RDCC-2035: Using PRD S2S Token when calling UP

### PR DESCRIPTION
**Pull request by AdilOoze**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-2035

### Change description ###

Using PRD S2S Token when calling UP

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by shashank-rastogi — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/professionalapi/configuration/FeignInterceptorConfiguration.java [hunk 1]
```diff
@@ -31,6 +38,9 @@ public RequestInterceptor requestInterceptor(FeignHeaderConfig config) {
                         String name = headerNames.nextElement();
                         String value = request.getHeader(name);
                         if (config.getHeaders().contains(name.toLowerCase())) {
+                            if (name.equalsIgnoreCase(AUTHORISATION)) {
+                                value = s2sClient.signIntoS2S();
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/configuration/FeignInterceptorConfiguration.java:42]: dont need to create new class /functionality for S2S . Just try below line and see if it works works.
authTokenGenerator will be instantiated with PRD S2S credentials from vault when sprint boot starts.

value  = authTokenGenerator.generate();

**Review by shreedhar-hmcts — COMMENTED** [review]
please check comment

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/S2sClient.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
+package uk.gov.hmcts.reform.professionalapi.controller;
+
+import static org.springframework.http.HttpHeaders.CONTENT_TYPE;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+import com.google.common.collect.ImmutableMap;
+import com.warrenstrange.googleauth.GoogleAuthenticator;
+
+import io.restassured.RestAssured;
```
Comment (SujeethKumarAri) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/S2sClient.java:9]: Correct me if I am wrong here. I think RestAssured is used for testing REST services right ?

Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/S2sClient.java]: I believe you are correct, the class has been removed in this commit: https://github.com/hmcts/rd-professional-api/pull/910/commits/c30848e3bc7dbe2db43a99dae26ffd39a2ecc9cf

File: src/main/java/uk/gov/hmcts/reform/professionalapi/configuration/FeignInterceptorConfiguration.java (hunk 1 above)
Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/configuration/FeignInterceptorConfiguration.java]: The changes have been made in this commit: https://github.com/hmcts/rd-professional-api/pull/910/commits/c30848e3bc7dbe2db43a99dae26ffd39a2ecc9cf

**Review by SujeethKumarAri — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/OrganisationInternalController.java [hunk 3]
```diff
@@ -137,7 +137,7 @@
     @GetMapping(produces = APPLICATION_JSON_VALUE)
     public ResponseEntity<Object> retrieveOrganisations(
             @Pattern(regexp = ORGANISATION_IDENTIFIER_FORMAT_REGEX, message = ORG_ID_VALIDATION_ERROR_MESSAGE)
-            @ApiParam(name = "id", required = false) @RequestParam(value = "id", required = false) String id,
+            @ApiParam(name = "id") @RequestParam(value = "id", required = false) String id,
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/OrganisationInternalController.java:140]: is this part of S2S changes ? I think this PR should be only for S2S changes , for any minor controller level changes, complete regression testing will be needed.

Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/OrganisationInternalController.java:140]: This is to address sonar smells, `required = false` is the default for `@ApiParam`

Comment (shreedhar-hmcts) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/OrganisationInternalController.java:140]: Got to know these are Sonar fixes, please have regression tested in PR before merging in master

**Review by shreedhar-hmcts — COMMENTED** [review]
Code looks fine. Once regressions testing for delete API is done, I will approve.

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/OrganisationInternalController.java (hunk 3 above)
Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/OrganisationInternalController.java:140]: have requested @mallikarjunputtana to regression test before merge

**Pull request closed without merging**